src/classes/AggregateResultProxy.cls

```diff
@@ -0,0 +1,16 @@
+public class AggregateResultProxy {
```

Comment:
I like your use of this proxy here. It's done this way so it supports aggregate queries of any `sObject` right ?

Reply:
Yep, exactly! I mainly wanted this proxy in order to support being able to mock methods that would normally accept an AggregateResult object or AggregateResult list as a param since we can't instantiate an AggregateResult like other sObjects.